qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java

```diff
@@ -544,36 +544,66 @@ public Ask generateAskGroupUsingBaseEntity(BaseEntity baseEntity) {
 		entityMessage.setReplace(true);
 
 		// create a child ask for every valid atribute
-		baseEntity.getBaseEntityAttributes().stream()
-				.filter(ea -> defBE.containsEntityAttribute("ATT_" + ea.getAttributeCode()))
+		defBE.getBaseEntityAttributes().stream()
+//		baseEntity.getBaseEntityAttributes().stream()
+//				.filter(ea -> defBE.containsEntityAttribute("ATT_" + ea.getAttributeCode()))
 				.forEach((ea) -> {
+					String attributeCode = StringUtils.removeStart(ea.getAttributeCode(),"ATT_");
+					attributeCode = StringUtils.removeStart(attributeCode,"DEP_");
+					attributeCode = StringUtils.removeStart(attributeCode,"DFT_");
+					attributeCode = StringUtils.removeStart(attributeCode,"UNQ_");
+
+					Optional<EntityAttribute> baseEA = baseEntity.findEntityAttribute(attributeCode);
+
+					if (!baseEA.isPresent()) {
+						String questionCode = attributeCode;
+						questionCode = "QUE_" + StringUtils.removeStart(StringUtils.removeStart(questionCode, "PRI_"), "LNK_");
+						Question childQues = null;
+
+						try {
+							Attribute attribute = databaseUtils.findAttributeByCode(userToken.getRealm(), attributeCode);
+							childQues = new Question(questionCode, attribute.getName(), attribute);
+						} catch (Exception ex) {
+							childQues = new Question(questionCode, ea.getAttribute().getName(), ea.getAttribute());
+						}
 
-					String questionCode = "QUE_"
-							+ StringUtils.removeStart(StringUtils.removeStart(ea.getAttributeCode(), "PRI_"), "LNK_");
+						Ask childAsk = new Ask(childQues, userToken.getUserCode(), attributeCode);
```

Comment:
The attribute code should not be the targetCode. It is also probably best if the sourceCode and targetCode are moved to variables since they are to be used multiple times in this piece of code.

Reply:
used base entity code as target code